### PR TITLE
feat: add doc for telemetry and usage data

### LINKED ISSUE #3788: Investigate telemetry errors

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

The telemetry investigation in #3788 led to a couple issues being opened to improve the events we're sending, but one of the comments was that we don't really have this written down anywhere.

Some other projects contain a USAGE_DATA.md file, so this is my attempt to both write down what we collect and why for users, but also start some coding notes for committers. Feedback appreciated.

### Screenshot/screencast of this PR

N/A

### What issues does this PR fix or reference?

Fixes #3788.

### How to test this PR?

N/A

### CODE REVIEW

File: USAGE_DATA.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+Telemetry should never be collected 'because we can', but because we want to see current behaviour in order to improve it. This means that we should only collect the minimal data in order to meet those goals, but conversely that we also collect enough to be able to act. For instance, collecting command pass/fail would tell us how often it failed, but not why it's failing in order to fix a bug or improve usability.
+
+When we capture information about commands and whether they succeed, they should be sent using logUsage() as a single event after an action completes, with optional error attributes. This makes it much easier to track the success rate and correlate failures with differences in environment or usage. logError() is typically used for independent or background errors.
+
+When calling logUsage() after an external commandline process or REST interface has failed, include an error object that contains the standard properties, e.g.:
+
+```
+error {
+  message : string
+  statusCode : int - for HTTP calls
```
Comment (benoitf) [inline USAGE_DATA.md]:
```suggestion
  statusCode : number, // for HTTP calls
```

File: USAGE_DATA.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  exitCode : int - for processes
```
Comment (benoitf) [inline USAGE_DATA.md]:
```suggestion
  exitCode : number, // for processes
```

File: USAGE_DATA.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Data Collection
+
+Podman Desktop uses telemetry to collect anonymous usage data in order to identify issues and improve our user experience. You can read our privacy statement
+[here](https://developers.redhat.com/article/tool-data-collection).
+
+Users are prompted during first startup to accept or decline telemetry. This setting can be
+changed at any time in Settings > Preferences > Telemetry.
+
+On disk this is stored in the `"telemetry.*"` keys within the settings file,
```
Comment (benoitf) [inline USAGE_DATA.md]: Maybe replace 'this' by setting as it's not where we store telemetry data but only the setting

File: USAGE_DATA.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+at `~/.local/share/containers/podman-desktop/configuration/settings.json`.
```
Comment (benoitf) [inline USAGE_DATA.md]: Maybe ~ --> HOME or %HOME%/$HOME

File: USAGE_DATA.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+## What's included in the telemetry data
+
+- General information, including operating system, machine architecture, and country.
+- When Podman Desktop starts and stops.
+- Which extensions are installed.
+- Which providers are used, and counts of objects like containers and images.
+- Pages navigated to in the UI.
+- Actions taken in the UI and interactions with providers, whether they succeeded, and how long they took.
+
+No personally identifiable information is captured. An anonymous id is used so that we can correlate the actions of a user even if we can't tell who they are.
```
Comment (benoitf) [inline USAGE_DATA.md:22]: should we include where anonymous ID is stored ?

File: USAGE_DATA.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+Likewise, content like error messages often contains user-identifiable data like folder names. These must also be stripped or hashed.
+
+### What to collect
+
+Telemetry should never be collected 'because we can', but because we want to see current behaviour in order to improve it. This means that we should only collect the minimal data in order to meet those goals, but conversely that we also collect enough to be able to act. For instance, collecting command pass/fail would tell us how often it failed, but not why it's failing in order to fix a bug or improve usability.
+
+When we capture information about commands and whether they succeed, they should be sent using logUsage() as a single event after an action completes, with optional error attributes. This makes it much easier to track the success rate and correlate failures with differences in environment or usage. logError() is typically used for independent or background errors.
+
```
Comment (benoitf) [inline USAGE_DATA.md:44]: should it include the import to use ? To call logUsage ?

File: USAGE_DATA.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+When calling logUsage() after an external commandline process or REST interface has failed, include an error object that contains the standard properties, e.g.:
+
+```
+error {
+  message : string
```
Comment (benoitf) [inline USAGE_DATA.md]:
```suggestion
  message : string,
```

File: USAGE_DATA.md (hunk 1 above)
Comment (deboer-tim) [reply USAGE_DATA.md]: Fixed in new commit.

File: USAGE_DATA.md (hunk 2 above)
Comment (deboer-tim) [reply USAGE_DATA.md]: Fixed in new commit.

File: USAGE_DATA.md (hunk 7 above)
Comment (deboer-tim) [reply USAGE_DATA.md]: Fixed in new commit.

File: USAGE_DATA.md (hunk 3 above)
Comment (deboer-tim) [reply USAGE_DATA.md]: Fixed in new commit.

File: USAGE_DATA.md (hunk 4 above)
Comment (deboer-tim) [reply USAGE_DATA.md]: Good call, used $HOME in new commit.

File: USAGE_DATA.md (hunk 5 above)
Comment (deboer-tim) [reply USAGE_DATA.md:22]: Mixed feelings about this, but I think it's only fair to list everything. Added in new commit.

File: USAGE_DATA.md (hunk 6 above)
Comment (deboer-tim) [reply USAGE_DATA.md:44]: Added in new commit.

**Review by benoitf — APPROVED** [review]

**Pull request merged**